src/shared/stores/__tests__/base.tests.js

```diff
@@ -0,0 +1,335 @@
+import * as mobx from 'mobx';
+import base from '../index';
+import menuSettings from './menuSettings.json';
+import listsFromMenu from './listsFromMenu.json';
+import slotsSettings from './slotsSettings.json';
+import slotsForItem from './slotsForItem.json';
+import slotsForColumn from './slotsForColumn.json';
+
+describe('Theme › Shared › Stores › Base', () => {
+  test('Has `lang`, `menu`, `commentsMap`, `scroll`, `share` and `shareModal` props', () => {
+    const self = base.create();
+
+    expect(self.lang).not.toBe(undefined);
+    expect(self.menu).not.toBe(undefined);
+    expect(self.commentsMap).not.toBe(undefined);
+    expect(self.scroll).not.toBe(undefined);
+    expect(self.share).not.toBe(undefined);
+    expect(self.shareModal).not.toBe(undefined);
+  });
+
+  // Don't know how to implement this test.
+  // test('`root` calls `getParent` with `self`', () => {});
+
+  test('`listsFromMenu` returns the menu from settings filtering anything that is not a list', () => {
+    const self = base.create();
+
+    Object.defineProperty(self, 'root', {
```

Comment:
I like this idea. I used to create a complex Store with root an and initial snapshot containing `build` or `settings`. This is cleaner and simpler, and it follows the principles of dependency injection.

cc: @DAreRodz